extensions/amp-story-subscriptions/0.1/amp-story-subscriptions.js

```diff
@@ -106,6 +124,14 @@ export class AmpStorySubscriptions extends AMP.BaseElement {
       StateProperty.SUBSCRIPTIONS_DIALOG_UI_STATE,
       (showDialog) => this.onSubscriptionsDialogUiStateChange_(showDialog)
     );
+
+    const ampSubscriptionsEl =
+      this.element.parentElement.parentElement.querySelector(
```

Comment:
Why not put the listener on `i-amphtml-story-subscriptions-dialog-banner-button`?

Reply:
So good question. This is because this entire dialog built under <amp-story-subscriptions> won't be displayed at all. The amp-subscriptions copies this and attach it to its <amp-subscriptions-dialog> and hides the original one. So adding event listener to this element won't do anything, and this is why we attach the event listener to the <amp-subscriptions-dialog>.